Fix non-deterministic behavior in legacy findFundamentalMat with USAC methods

Jatin-Y07 opened the pull request:
### Summary
This PR fixes a regression reported in Issue #27388 where `findFundamentalMat` produces non-deterministic results when using `USAC` methods (e.g., `USAC_MAGSAC`) in recent OpenCV versions.

### The Problem
In OpenCV 4.11+, the default `UsacParams` enable parallel execution (`isParallel = true`) and use a random seed. The legacy overload of `findFundamentalMat` (which takes `int method`) delegates to these defaults, causing the output to vary between runs even with identical inputs. This breaks behavior compatibility with older versions (e.g., 4.6.0) where results were deterministic.

### The Fix
This patch modifies the legacy wrapper in `modules/calib3d/src/fundam.cpp`. When a `USAC` method is selected via the legacy API, we now explicitly enforce:
- `randomGeneratorState = 0`: Ensures a fixed seed is used.
- `isParallel = false`: Disables race conditions from parallel RANSAC to guarantee bit-exact reproducibility.

Jatin-Y07 committed:
Update fundam.cpp

Jatin-Y07 committed:
issue #27388 fixed with a simple test added.

This PR have the fix of issue #27388 and a simple test in test_fundam.cpp that highlights the issue.
Thanks

opencv-alalek commented on modules/calib3d/src/fundam.cpp:
```diff
@@ -856,8 +856,43 @@ cv::Mat cv::findFundamentalMat( InputArray _points1, InputArray _points2,
+
+        // Fix for Issue #27388: Restore deterministic behavior
+        params.isParallel = false;
+        params.randomGeneratorState = (int)cv::theRNG();
```
No need to **generate** random numbers:

```
    std::cout << (int)cv::theRNG() << std::endl;
    std::cout << (int)cv::theRNG() << std::endl;
```

Store RNG state or `cv::RNG` itself (no way without API changes, so unlikely).

asmorkalov replied:
Friendly reminder.

Jatin-Y07 replied:
I have tried to apply the fix can you check and tell me if it still needs to be changed.
Thanks, and sorry for delay

Jatin-Y07 committed:
New fix for #27388

This commit fixes Issue #27388 where legacy USAC methods in findFundamentalMat produced non-deterministic results.

Changes:
1. Restore determinism: Explicitly set `isParallel = false` and initialize `randomGeneratorState` using the global `cv::theRNG().state`.
2. Fix RNG usage: Read the RNG state directly without casting/calling the object, preventing side effects (advancing the seed) and ensuring strict reproducibility for consecutive calls.
3. Fix compilation error: Replaced the internal helper call with explicit parameter mapping to resolve the "no member named method" build failure.
4. Added regression test `Regression_27388_Deterministic` to verify bit-exact reproducibility.
